### PR TITLE
Bundle using Webpack

### PR DESCRIPTION
**Pull request by OmgImAlexis**
I know this is a crappy single commit with loads of changes but I just wanted feedback before I continue.

This should work out of the box in the dark theme. Do not run any gulp commands, etc. or it'll break as I've had to manually change a few things after the sync.

### CODE REVIEW

File: themes-default/slim/package.json [hunk 1]
```diff
@@ -7,6 +7,7 @@
     "gulp": "gulp",
     "lint": "xo",
     "build": "cross-env NODE_ENV=production gulp build",
+    "webpack": "webpack-cli && cp dist/js/*.js ../../themes/dark/assets/js && cp dist/js/*.js ../../themes/light/assets/js",
```
Comment (OmgImAlexis) [inline themes-default/slim/package.json]: You can just use `webpack` here.

File: themes-default/slim/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+    "compilerOptions": {
+        "allowJs": true,
+        "noImplicitAny": true
+    },
+    "include": [
+        "./static/js/*.js"
```
Comment (OmgImAlexis) [inline themes-default/slim/tsconfig.json]: This isn't including any subdirs?

Comment (sharkykh) [reply themes-default/slim/tsconfig.json]: That's your code, What's this even for?

Comment (OmgImAlexis) [reply themes-default/slim/tsconfig.json]: Oh shoot, it is.

It's for enabling typescript typechecking and intellisense in vscode. That allows us to use the jsdoc syntax to add types for all of our functions, etc. and then it'll use those to create our type file so we don't need todo it ourselves.

**Comment by OmgImAlexis** [comment]
Use this now. The `cp` wasn't being run for some reason.

```
cd themes-default/slim
yarn webpack && yarn webpack-copy && yarn gulp sync
```

**Comment by OmgImAlexis** [comment]
@sharkykh can we try and get this merged today?

File: themes-default/slim/package.json [hunk 3]
```diff
@@ -4,10 +4,10 @@
   "name": "slim",
   "author": "p0psicles",
   "scripts": {
-    "gulp": "gulp",
```
Comment (OmgImAlexis) [inline themes-default/slim/package.json]: We don't need this as `yarn gulp` is the same even if you don't have this.

**Comment by p0psicles** [comment]
Please don't forget to update the WIKI.
It get's overlooked every time. And after the PR is merged, nobody seems to have time to update it.

**Pull request merged**